lib/ionic-project-transformator.ts

```diff
@@ -0,0 +1,416 @@
+		return (() => {
+			let iconSourceFolder = <string>resource.src;
+
+			// The path to the resource will depend on what operating system the project was created on.
+			let resourceName = _.last(iconSourceFolder.indexOf("\\") >= 0 ? iconSourceFolder.split("\\") : iconSourceFolder.split("/"));
+
+			// Windows Phone 8 supports only JPG splash screens and Ionic provides PNG splash screens.
+			if (platformName === IonicProjectTransformator.WINDOWS_PHONE_PLATFORM_NAME) {
+				resourceName = resourceName.replace(IonicProjectTransformator.WINDOWS_PHONE_IONIC_SPLASH_SCREEN_FORMAT, IonicProjectTransformator.WINDOWS_PHONE_SUPPORTED_SPLASH_SCREEN_FORMAT);
+			}
+
+			if (platformName === IonicProjectTransformator.ANDROID_PLATFORM_NAME) {
+				platformConfigXmlDestinationDirectory = path.join(appBuilderResourcesDirectory, this.$projectConstants.APPBUILDER_PROJECT_PLATFORMS_NAMES[platformName], IonicProjectTransformator.ANDROID_XML_FOLDER_NAME);
+
+				// Ionic resource name will be <resourceType>-<orientationAndDensity>-<name>.
+				let androidResourceInformation = resourceName.split(`-${resource.density}-`);
+				let androidActualResourceName = _.last(androidResourceInformation);
+				let androidResourceType = _.first(androidResourceInformation);
+
+				resource.src = path.relative(path.join(platformConfigXmlDestinationDirectory), path.join(appBuilderPlatformResourcesDirectory, `${androidResourceType}-${resource.density}`, androidActualResourceName));
+			} else {
+				resource.src = resourceName;
+			}
+
+			return resource;
+		}).future<ConfigXmlFile.IResource>()();
+	}
+
+	private cloneResourcesCore(projectDir: string, appBuilderResourcesDirectory: string, platformKeyName: string, cloneFunction: Function): IFuture<void> {
+		return (() => {
+			let ionicPlatformName = this.$projectConstants.IONIC_PROJECT_PLATFORMS_NAMES[platformKeyName];
+			let appBuilderPlatformName = this.$projectConstants.APPBUILDER_PROJECT_PLATFORMS_NAMES[platformKeyName];
+
+			let ionicPlatformResourcesDirectory = path.join(this.ionicResourcesDirectory, ionicPlatformName);
+
+			if (!this.checkIfPlatformIsAddedToProject(this.ionicResourcesDirectory, ionicPlatformName).wait()) {
+				return;
+			}
+
+			let ionicPlatformResources = this.$fs.readDirectory(ionicPlatformResourcesDirectory).wait();
+
+			let appbuilderPlatformResourcesDirectory = path.join(appBuilderResourcesDirectory, appBuilderPlatformName);
+			if (!this.$fs.exists(appbuilderPlatformResourcesDirectory).wait()) {
+				this.$fs.createDirectory(appbuilderPlatformResourcesDirectory).wait();
+			}
+
+			_.each(ionicPlatformResources, (resourceName: string) => {
+				let resourceDirectory = path.join(ionicPlatformResourcesDirectory, resourceName);
+				let resources = this.$fs.readDirectory(resourceDirectory).wait();
+
+				_.each(resources, (ionicResourceName: string) => {
+					cloneFunction.apply(this, [resourceDirectory, appbuilderPlatformResourcesDirectory, ionicResourceName, resourceName]).wait();
```

Comment:
`cloneFunction(...[resourceDirectory, appbuilderPlatformResourcesDirectory, ionicResourceName, resourceName]).wait();`
will be transpiled to:
`cloneFunction.apply(void(0), [resourceDirectory, appbuilderPlatformResourcesDirectory, ionicResourceName, resourceName]).wait();`
and the context will not be saved.